lib/schema/manifest.go

```diff
@@ -169,7 +169,12 @@ func (m *Manifest) SetBase(locator loc.Locator) {
 // WithBase returns a copy of this manifest with the specified base application
 func (m *Manifest) WithBase(locator loc.Locator) Manifest {
 	result := *m
-	result.SetBase(locator)
+	if m.BaseImage != nil {
+		result.BaseImage = &BaseImage{Locator: locator}
+	}
+	result.SystemOptions = &SystemOptions{
```

Comment:
Won't this basically override the entire systemOptions field the manifest may have set, rather than just setting the runtime locator?

Reply:
It's a good catch - I'll update it conditionally.